### PR TITLE
add dockerfile and parameter builder for appdynamics collector

### PR DESCRIPTION
**Pull request by virtualpeter**

### CODE REVIEW

**Comment by MarkRx** [comment]
Please get this merged :)

**Comment by MarkRx** [comment]
Changes included with #1708

File: collectors/performance/appdynamics/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+FROM docker.io/openjdk:8-jre
+
+MAINTAINER Hygieia@capitalone.com
+
+
+ENV SPRING_DATA_MONGODB_DATABASE=dashboard
+ENV SPRING_DATA_MONGODB_HOST=10.0.1.1
+ENV SPRING_DATA_MONGODB_PORT=9999
+ENV SPRING_DATA_MONGODB_USERNAME=db
+ENV SPRING_DATA_MONGODB_PASSWORD=ENC(aSPTk36yA/ZklUg75CrZ8w==)
```
Comment (satishc1) [inline collectors/performance/appdynamics/docker/Dockerfile]: @virtualpeter Please take these out. The default values are set in the prop builder.

Comment (virtualpeter) [reply collectors/performance/appdynamics/docker/Dockerfile]: these lines are necessary to support docker-compose v2 and help when debugging this declares the variables so they can be passed in at runtime -  i followed the pattern set by all the other collectors' Dockerfile and prop builders in that they are defaulted here as well as in the collector.

Comment (asg247) [reply collectors/performance/appdynamics/docker/Dockerfile]: @virtualpeter the ENV variables can be passed through docker-compose.yml file.

File: collectors/performance/appdynamics/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  mkdir /hygieia
+
+COPY hygieia/ /hygieia
+COPY appdynamics-properties-builder.sh /hygieia/
+
+WORKDIR /hygieia
+
+VOLUME ["/hygieia/logs"]
+
+EXPOSE 8080
```
Comment (asg247) [inline collectors/performance/appdynamics/docker/Dockerfile]: Not required to expose any port for collectors

**Comment by asg247** [comment]
Couple of issues with this PR.

1. The appdynamics-properties-builder.sh does not has execute permission, it should be set as 755. Otherwise, it wont be executed when container starts.
2. logback.xml is missing.

I can submit a PR on top of this PR to fix all the issues.

**Comment by satishc1** [comment]
@tabladrum LGTM

**Pull request merged**